<launch>

  <!-- Load robot model -->
  <arg name="model" default="$(find vss_simulation)/urdf/generic_vss_robot.xacro"/>
  <arg name="debug" default="false"/>
  <arg name="gui" default="true"/>
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="world_name" default="vss_field.world"/>
  <arg name="recording" default="false"/>
  <arg name="x" default="-0.25" />
  <arg name="y" default="0" />
  <arg name="yaw" default="0" />
  <arg name="twist_interface" default="true"/>
  <arg name="config_file" default=""/>

  <!-- Control option -->
  <arg name="keyboard" default="false"/>

  <!-- Load gazebo simulation -->
  <include file="$(find vss_simulation)/launch/gazebo.launch">
    <arg name="debug" value="$(arg debug)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="recording" value="$(arg recording)"/>
  </include>

  <group ns="robot1">
    <include file="$(find vss_simulation)/launch/spawn_robot.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="x" value="$(arg x)" />
      <arg name="y" value="$(arg y)" />
      <arg name="yaw" value="$(arg yaw)" />
      <arg name="twist_interface" value="$(arg twist_interface)"/>
      <arg name="config_file" value="$(arg config_file)"/>
    </include>
  </group>

  <group if="$(arg keyboard)">
    <node name="keyboard_controller" pkg="vss_simulation" type="keyboard_node.py"/>
  </group>

  <group unless="$(arg keyboard)">

    <!-- Display controllers to user -->
    <node name="rqt_robot_steering" pkg="rqt_robot_steering" type="rqt_robot_steering">
      <param name="default_topic" value="/robot1/vss_robot_diff_drive_controller/cmd_vel"/>
    </node>

  </group>

</launch>
